srai/joiners/intersection_joiner.py

```diff
@@ -0,0 +1,67 @@
+"""
+Intersection Joiner.
+
+This module contains intersection joiner implementation.
+
+"""
+
+from dataclasses import dataclass
+
+import geopandas as gpd
+import pandas as pd
+
+
+@dataclass
+class JoinerResult:
+    """Joiner result dataclass."""
+
+    regions: gpd.GeoDataFrame
+    features: gpd.GeoDataFrame
+    joined: gpd.GeoDataFrame
```

Comment:
Maybe name this attribute `joint`? Or maybe make `joined` a more verbose name. It doesn't sound well IMO.